<h1>ActiveSupport::Rescuable::ClassMethods#rescue_from</h1>


<p>Rescue exceptions raised in controller actions.</p>

<p><tt>rescue_from</tt> receives a series of exception classes or class names,
and a trailing <tt>:with</tt> option with the name of a method or a Proc
object to be called to handle them. Alternatively a block can be given.</p>

<p>Handlers that take one argument will be called with the exception, so that
the exception can be inspected when dealing with it.</p>

<p>Handlers are inherited. They are searched from right to left, from bottom
to top, and up the hierarchy. The handler of the first class for which
<tt>exception.is_a?(klass)</tt> holds true is the one invoked, if any.</p>

<pre>class ApplicationController &lt; ActionController::Base
  rescue_from User::NotAuthorized, :with =&gt; :deny_access # self defined exception
  rescue_from ActiveRecord::RecordInvalid, :with =&gt; :show_errors

  rescue_from 'MyAppError::Base' do |exception|
    render :xml =&gt; exception, :status =&gt; 500
  end

  protected
    def deny_access
      ...
    end

    def show_errors(exception)
      exception.record.new_record? ? ...
    end
end</pre>





